models/resnet.py

```diff
@@ -83,26 +86,29 @@ def __init__(self, num, num_classes, groups=1, width_per_group=64):
     self.layer2 = self._make_layer(self.block, 128, self.num_blocks[1], stride=2)
     self.layer3 = self._make_layer(self.block, 256, self.num_blocks[2], stride=2)
     self.layer4 = self._make_layer(self.block, 512, self.num_blocks[3], stride=2)
-    self.fc = nn.Linear(512 * self.block.expansion, num_classes)
+    if num_classes is not None:
+      self.fc = nn.Linear(512 * self.block.expansion, num_classes)
```

Comment:
`self.fc = nn.Linear(512 * self.block.expansion, num_classes) if num_classes is not None else None`

Reply:
done